drivers/cc110x/cc110x.c

```diff
@@ -0,0 +1,393 @@
+    cc110x_writeburst_reg(dev, 0x00, cc110x_default_conf, cc110x_default_conf_size);
+
+    /* Write PATABLE (power settings) */
+    cc110x_writeburst_reg(dev, CC110X_PATABLE, CC110X_DEFAULT_PATABLE, 8);
+
+    /* set base frequency */
+    cc110x_set_base_freq_raw(dev, CC110X_DEFAULT_FREQ);
+
+    /* Set default channel number */
+    cc110x_set_channel(dev, CC110X_DEFAULT_CHANNEL);
+
+    /* set default node id */
+#ifdef CPUID_ID_LEN
+    if (CPUID_ID_LEN>0) {
+        char cpuid[CPUID_ID_LEN];
+        cpuid_get(cpuid);
+        cc110x_set_address(dev, (uint8_t) cpuid[CPUID_ID_LEN-1]);
+    }
+#endif
+
+    LOG_INFO("cc110x: initialized with address=%u and channel=%i\n",
+            (unsigned)dev->radio_address,
+            dev->radio_channel);
+
+    return 0;
+}
+
+uint8_t cc110x_set_address(cc110x_t *dev, uint8_t address)
+{
+    DEBUG("%s:%s:%u setting address %u\n", RIOT_FILE_RELATIVE, __func__,
+            __LINE__, (unsigned)address);
+    if (!(address < MIN_UID) || (address > MAX_UID)) {
+        if (dev->radio_state != RADIO_UNKNOWN) {
+            cc110x_write_register(dev, CC110X_ADDR, address);
+            dev->radio_address = address;
+            return address;
+        }
+    }
+
+    return 0;
+}
+
+void cc110x_set_base_freq_raw(cc110x_t *dev, const char* freq_array)
+{
+#if ENABLE_DEBUG == 1
+    uint8_t _tmp[] = { freq_array[2], freq_array[1], freq_array[0], 0x00};
+    uint32_t *FREQ = (uint32_t*) _tmp;
+
+    DEBUG("cc110x_set_base_freq_raw(): setting base frequency to %uHz\n",
+            (26000000>>16) * (unsigned)(*FREQ));
+#endif
+    cc110x_writeburst_reg(dev, CC110X_FREQ2, freq_array, 3);
+}
+
+void cc110x_set_monitor(cc110x_t *dev, uint8_t mode)
+{
+    DEBUG("%s:%s:%u\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+
+    cc110x_write_register(dev, CC110X_PKTCTRL1, mode ? 0x04 : 0x06);
+}
+
+void cc110x_setup_rx_mode(cc110x_t *dev)
+{
+    DEBUG("%s:%s:%u\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+
+    /* Stay in RX mode until end of packet */
+    cc110x_write_reg(dev, CC110X_MCSM2, 0x07);
+    cc110x_switch_to_rx(dev);
+}
+
+void cc110x_switch_to_rx(cc110x_t *dev)
+{
+    DEBUG("%s:%s:%u\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+
+#ifdef MODULE_CC110X_HOOKS
+    cc110x_hook_rx();
+#endif
+
+    gpio_irq_disable(dev->params.GDO2);
+
+    /* flush RX fifo */
+    cc110x_strobe(dev, CC110X_SIDLE);
+    cc110x_strobe(dev, CC110X_SFRX);
+
+    dev->radio_state = RADIO_RX;
+
+    cc110x_write_reg(dev, CC110X_IOCFG2, 0x6);
+    cc110x_strobe(dev, CC110X_SRX);
+
+    gpio_irq_enable(dev->params.GDO2);
+}
+
+void cc110x_wakeup_from_rx(cc110x_t *dev)
+{
+    DEBUG("%s:%s:%u\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+
+    if (dev->radio_state != RADIO_RX) {
+        return;
+    }
+
+    LOG_DEBUG("cc110x: switching to idle mode\n");
+
+    cc110x_strobe(dev, CC110X_SIDLE);
+    dev->radio_state = RADIO_IDLE;
+}
+
+char *cc110x_get_marc_state(cc110x_t *dev)
```

Comment:
It's indeed called `MARCSTATE` in the cc11xx (MAin Radio Control State Machine State).